CMakeLists.txt

```diff
@@ -9,7 +8,71 @@ set(BUILD_SHARED_LIBS ON)
 set(CMAKE_CXX_STANDARD 17)
 
 include_directories(${CMAKE_SOURCE_DIR}/ers)
+include_directories(${CMAKE_SOURCE_DIR})
 
 add_subdirectory(bin)
 add_subdirectory(src)
 add_subdirectory(test)
+
+# Installation stuff copied from the double-conversion CMakeLists.txt, which in turn got it from
+# https://github.com/forexample/package-example
+
+include(GNUInstallDirs)
+
+# Layout. This works for all platforms:
+#   * <prefix>/lib/cmake/<PROJECT-NAME>
+#   * <prefix>/lib/
+#   * <prefix>/include/
+set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+
+set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
+
+# Configuration
+set(version_config "${generated_dir}/${PROJECT_NAME}ConfigVersion.cmake")
+set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
+set(targets_export_name "${PROJECT_NAME}Targets")
+set(namespace "${PROJECT_NAME}::")
+
+# Include module with function 'write_basic_package_version_file'
+include(CMakePackageConfigHelpers)
+
+# Configure '<PROJECT-NAME>ConfigVersion.cmake'
+# Note: PROJECT_VERSION is used as a VERSION
+write_basic_package_version_file(
+    "${version_config}" COMPATIBILITY SameMajorVersion
+)
+
+# Configure '<PROJECT-NAME>Config.cmake'
+# Use variables:
+#   * targets_export_name
+#   * PROJECT_NAME
+configure_package_config_file(
+    "cmake/Config.cmake.in"
+    "${project_config}"
+    INSTALL_DESTINATION "${config_install_dir}"
+)
+
+install(
+    TARGETS ers ErsBaseStreams config
```

Comment:
@philiprodrigues this line gives me an error:

```
$ cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/opt/ers
-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   regex
/usr/lib/x86_64-linux-gnu/libboost_regex.so
CMake Error at CMakeLists.txt:55 (install):
  install TARGETS given target "ers" which does not exist in this directory.
```

CMake is still largely a mystery to me but maybe this install target needs to be in `src/CMakeLists.txt`?

Reply:
Ah yes. Apparently cmake 3.13 changed that behaviour so that what I wrote is allowed. I have a fix locally so it should work with cmake < 3.13 but I got myself into branch/fork hell which will have to wait till tomorrow to sort out.

https://stackoverflow.com/questions/34443128/cmake-install-targets-in-subdirectories/34444949#34444949

Reply:
Fixed in PR #6